android/modules/ui/src/java/ti/modules/titanium/ui/NavigationWindowProxy.java

```diff
@@ -47,7 +50,7 @@ public void popToRootWindow(@Kroll.argument(optional = true) Object arg)
 	{
 		// Keep first "root" window
 		for (int i = windows.size() - 1; i > 0; i--) {
-			WindowProxy window = windows.get(i);
+			WindowProxy window = ((WindowProxy) windows.get(i));
```

Comment:
Casting it to `WindowProxy` will crash if it's of type `TabGroup`. Since you've changed the `windows` member variable to be a collection of `TiWindowProxy` types, you can get rid of the cast and close the `TiWindowProxy` object.

You will also need to change the `NavigationWindowProxy.closeWindow()` method so that it can close a tab group. Perhaps change the `closeWindow()` argument from type `WindowProxy` to `Object` and then do an `instanceof TiWindowProxy` like how we're doing it in the `openWindow()` method.